react/week4/GitHub /github-user-search/src/User.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react'
+import SearchBox from './SearchBox'
+
+const User = () => {
+
+    const[input, setInput] = useState('')
+    const[isLoading,setIsLoading]=useState(false)
+    const[error,setError]=useState(false)
+    const[users, setUsers] = useState([])
+
+    useEffect(() => {
+        
+        if(!input){
+           return
+        }
+         setIsLoading(true)
+        
+        fetch (`https://api.github.com/search/users?q=${input}`)
+        .then(response =>  response.json())
+        .then(data => {
+            console.log(data)
+            setIsLoading(false)
+            setUsers(data.items)
+        })
+    .catch(err => {
+        setError(true)
+        setIsLoading(false)
+        console.error(err)
+    })
+},[input])
+   
+
+const submitHandler = e => {
+    e.preventDefault()
+    setInput(e.target.elements.query.value)
```

Comment:
might be more readable if you saved the search query in state (with useState) inside `SearchBox` and then `onSubmit` you send the saved query back as an argument to `submitHandler`.

What you did works, but downside is that you reference the name of the field in here (`query`), which is defined in `SearchBox`. Having to match the name in two different files makes it harder to maintain the code.